src/components/MultiHopTrade/components/SharedTradeInput/SharedTradeInputHeader.tsx

```diff
@@ -21,6 +23,7 @@ export const SharedTradeInputHeader = ({
 
   const enableBridgeClaims = useFeatureFlag('ArbitrumBridgeClaims')
   const enableLimitOrders = useFeatureFlag('LimitOrders')
+  const walletId = useAppSelector(selectWalletId)
```

Comment:
[non-blocking]: Could possibly be more idiomatic to use `selectIsWalletConnected` rather than using the presence of a walletId (i.e does the presence of a wallet ID in redux mean the wallet is actually connected?)

Reply:
Conceptually agree and also thought we could use `isConnected` from `useWallet()` instead (not only here but across the board with other usages too), however, for most cases `walletId` makes more sense IMO, because MM or KK could be "disconnected" (i.e warning in `<ConnectedMenu />` on the top-right), but you're still in the context of having a wallet, just in disconnected state if that makes sense 🤯 and claim button would disappear when we're in such state if we used that